include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,183 @@
+            struct linear_distribute_floating_point_view
+              : view_facade<linear_distribute_floating_point_view<T>, finite>
+            {
+                static_assert(std::is_floating_point<T>(), "");
+            private:
+                friend range_access;
+
+                T from_, to_;
+                std::ptrdiff_t n_;
+                
+                constexpr T read() const
+                {
+                    return from_;
+                }
+                constexpr bool equal(default_sentinel) const
+                {
+                    return n_ == 0;
+                }
+                RANGES_CXX14_CONSTEXPR void next()
+                {
+                    RANGES_EXPECT(n_ > 0);
+                    from_ += (to_ - from_) / T(--n_);
+                }
+                RANGES_CXX14_CONSTEXPR void prev()
+                {
+                    from_ -= (to_ - from_) / T(++n_);
+                }
+            public:
+                linear_distribute_floating_point_view() = default;
+                linear_distribute_floating_point_view(T from, T to, std::ptrdiff_t n)
+                  : from_(from), to_(to), n_(n)
+                {
+                    RANGES_EXPECT(n_ > 0);
+                    RANGES_EXPECT(to_ >= from_);
+                }
+                constexpr std::size_t size() const noexcept
+                {
+                    return static_cast<std::size_t>(n_);
+                }
+            };
+
+        } // namespace detail
+
+        /// \endcond
+
+        /// \addtogroup group-views
+        /// @{
+
+        /// Distributes `n >= 2` values linearly in the closed interval [`from`, `to`].
+        template <typename T>
+        using linear_distribute_view
+        = meta::if_c<std::is_floating_point<T>::value,
+                     detail::linear_distribute_floating_point_view<T>,
+                     detail::linear_distribute_integral_view<T>>;
+        ///@}
+
+        namespace view
+        {
+            /// Distributes `n >= 2` values linearly in the closed interval `[from, to]`.
+            struct linear_distribute_fn
+            {
+                template<typename T,
+                    CONCEPT_REQUIRES_(std::is_arithmetic<T>())>
+                constexpr auto operator()(T from, T to, std::ptrdiff_t n) const
+                RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT
+                (
+                    linear_distribute_view<T>{from, to, n}
+                )
+
+            #ifndef RANGES_DOXYGEN_INVOKED
+                template<typename T,
+                    CONCEPT_REQUIRES_(not std::is_arithmetic<T>())>
+                void operator()(T, T, std::ptrdiff_t) const
+                {
+                    CONCEPT_ASSERT_MSG(std::is_arithmetic<T>(),
+                        "The object passed to view::linear_distribute must model the Arithmetic "
+                        "concept; that is, it must be one of the native integer of floating-point types");
```

Comment:
I'm with Casey on this, we should have a solid concept for user-defined "integers" before trying to bolt it on here. Having a couple of views that require `std::is_integral` (or `std::is_arithmetic`), will help getting that concept right.